test/test_parser.cpp

```diff
@@ -5208,4 +5208,27 @@ TEST_TYPES(Parser_Arithmetic, Prop<int64_t>, Prop<float>, Prop<double>, Prop<Dec
     verify_query_sub(test_context, person, "age * $0 == $1", args, 1);
 }
 
+TEST(Parser_Between)
+{
+    Group g;
+    TableRef table = g.add_table("table");
+    auto col_age = table->add_column(type_Int, "age");
+    auto col_scores = table->add_column_list(type_Int, "scores");
+    for (int i = 0; i < 3; ++i) {
+        auto list = table->create_object().set(col_age, 5 * i + 30).get_list<Int>(col_scores);
+        for (int j = 0; j < 5; j++) {
+            list.add(j + i * 5);
+        }
+        for (int j = 0; j < 5; j++) {
+            list.add(j + i * 5 + 10);
+        }
+    }
+    g.to_json(std::cout);
+    verify_query(test_context, table, "age between {30, 37}", 2);
```

Comment:
I think we need to throw on `NONE age between {30, 37}` as well?

Reply:
We do already. I added a test for it